public/app/features/commandPalette/actions/dashboardActions.ts

```diff
@@ -42,7 +44,7 @@ export async function getRecentDashboardActions(): Promise<CommandPaletteAction[
   return recentDashboardActions;
 }
 
-export async function getDashboardSearchResultActions(searchQuery: string): Promise<CommandPaletteAction[]> {
+export async function getDashboardSearchResultActions(searchQuery: string): Promise<ActionImpl[]> {
```

Comment:
I think this should still continue to return `CommandPaletteAction`, and we can map it to `ActionImpl` inside of `RenderResults`